src/debug_utils.cc

```diff
@@ -0,0 +1,226 @@
+#include "debug_utils.h"
+#include "node_internals.h"
+
+#ifdef __POSIX__
+#if defined(__linux__)
+#include <features.h>
+#endif
+
+#if defined(__linux__) && !defined(__GLIBC__) || \
+    defined(__UCLIBC__) || \
+    defined(_AIX)
+#define HAVE_EXECINFO_H 0
+#else
+#define HAVE_EXECINFO_H 1
+#endif
+
+#if HAVE_EXECINFO_H
+#include <cxxabi.h>
+#include <dlfcn.h>
+#include <execinfo.h>
+#include <unistd.h>
+#include <sys/mman.h>
+#include <stdio.h>
+#endif
+
+#else  // __POSIX__
+
+#include <windows.h>
+#include <dbghelp.h>
+
+#endif  // __POSIX__
+
+namespace node {
+
+#ifdef __POSIX__
+#if HAVE_EXECINFO_H
+class PosixSymbolDebuggingContext final : public NativeSymbolDebuggingContext {
+ public:
+  PosixSymbolDebuggingContext() : pagesize_(getpagesize()) { }
+
+  SymbolInfo LookupSymbol(void* address) override {
+    Dl_info info;
+    const bool have_info = dladdr(address, &info);
+    SymbolInfo ret;
+    if (!have_info)
+      return ret;
+
+    if (info.dli_sname != nullptr) {
+      if (char* demangled = abi::__cxa_demangle(info.dli_sname, 0, 0, 0)) {
+        ret.name = demangled;
+        free(demangled);
+      } else {
+        ret.name = info.dli_sname;
+      }
+    }
+
+    if (info.dli_fname != nullptr) {
+      ret.filename = info.dli_fname;
+    }
+
+    return ret;
+  }
+
+  bool IsMapped(void* address) override {
+    void* page_aligned = reinterpret_cast<void*>(
+        // NOLINTNEXTLINE(whitespace/operators)
+        reinterpret_cast<uintptr_t>(address) &~ (pagesize_ - 1));
```

Comment:
Why not `... & ~(...)`?  Then you don't have to suppress the linter.

Reply:
I’m getting a déjà vu with that review comment … :smile: I’ve switched it for now, but I really prefer `&~` for masking operations